### ISSUE TITLE
add checkers for URL-like fields

### ISSUE THREAD

**Issue by HubMiner** [keepright/keepright#85]
Please consider adding checkers for fields that contain full or partial URL:
wikipedia
wikidata
contact:*=http*
*facebook*=
*yelp*=
*tripadvisor*=
*url*=

**Comment by bkil** [comment]
I found some more using various TagInfo searches and https://wiki.openstreetmap.org/wiki/Key:contact , some of these overlap with your suggestions (like `contact:*=http`):
`mapillary, flickr, image:url, source:image, contact:webcam (url:webcam, webcam), contact:facebook, contact:instagram, contact:twitter, contact:youtube, contact:ok, contact:telegram, contact:whatsapp, contact:linkedin, contact:pinterest, contact:viber (?), contact:foursquare, contact:skype (?), contact:xing, contact:flickr, contact:mastodon, contact: diaspora, contact:gnusocial, contact:messenger, wikimedia_commons, wiki:symbol, *:source, source:*, source_ref, source_ref:*, link.`

Just looking at the length of this list, maybe we should keep a typed list of keys where we expect and validate a structured URL (or either a URL or an ID) and maybe implement an automated tool to update this list of keys.

**Comment by HubMiner** [comment]
Additional scope: I came across contact:facebook value that was not a complete URL.  The problem is that the keyword was not easily expandable - real URL was a facebook-custom id.  It would be great if this scanned common contact:SITE keys (as bkil nicely spelled out), and it would report the values that are not proper URLs.

**Comment by bkil** [comment]
Yes, this is usually the case with many other keys, like `contact:viber`, `wikimedia_commons`, `contact:skype` and some others as well - they can have two or more alternative valid syntax. However, I think both the long URL form and the short ID form should be validated and accepted.

It is an advantage to support shorter forms, as it has just been mentioned on the OSM tagging list that more values could fit into the 255 character limit this way. It is also easier on the eyes of human mappers, while still being accessible to machines.

I think it can be trivially expanded - it just needs a few more lines of normalization before checking. As most of those that I've seen have either a constant length and character set or is a username, validation is like checking for conformance to `[0-9a-f]{23}` and they can then be simply expanded to a full URL by prefixing with a string (`printf ...%s...` at worst).

A few hard coded values need to be skipped, though, like `yes`, and `no`.

Anyway we probably don't need to do much validation - if downloading works after inserting a prefix, we're good - maybe even `[^/]+` could also suffice.
